creator-node/src/services/stateMachineManager/stateReconciliation/issueSyncRequest.jobProcessor.js

```diff
@@ -43,48 +44,94 @@ const mergePrimaryAndSecondaryEnabled = config.get(
  * @param {Object} param.syncRequestParameters axios params to make the sync request. Shape: { baseURL, url, method, data }
  */
 module.exports = async function ({
-  logger,
   syncType,
   syncMode,
-  syncRequestParameters
+  syncRequestParameters,
+  logger = genericLogger
 }) {
-  _validateJobData(logger, syncType, syncMode, syncRequestParameters)
+  let jobsToEnqueue = {}
+  let metricsToRecord = []
+  let error = {}
 
-  const isValidSyncJobData =
-    'baseURL' in syncRequestParameters &&
-    'url' in syncRequestParameters &&
-    'method' in syncRequestParameters &&
-    'data' in syncRequestParameters
-  if (!isValidSyncJobData) {
-    const errorMsg = `Invalid sync data found: ${JSON.stringify(
-      syncRequestParameters
-    )}`
-    logger.error(errorMsg)
-    return {
-      error: {
-        message: errorMsg
-      },
-      jobsToEnqueue: {}
-    }
+  const startTimeMs = Date.now()
+
+  const {
+    syncReqToEnqueue: syncReqToEnqueueResp,
+    result,
+    error: errorResp
+  } = await _handleIssueSyncRequest({
+    syncType,
+    syncMode,
+    syncRequestParameters,
+    logger
+  })
+  if (errorResp) {
+    error = errorResp
+    logger.error(error.message)
   }
-  if (
-    !(syncRequestParameters.data.wallet instanceof Array) ||
-    !syncRequestParameters.data.wallet?.length
-  ) {
-    const errorMsg = `Invalid sync data wallets (expected non-empty array): ${syncRequestParameters.data.wallet}`
-    logger.error(errorMsg)
-    return {
-      error: {
-        message: errorMsg
-      },
-      jobsToEnqueue: {}
+
+  // Determine if new sync request needs to be enqueued
+  if (syncReqToEnqueueResp) {
+    const {
+      userWallet,
+      secondaryEndpoint,
+      syncMode: syncModeResp
+    } = syncReqToEnqueueResp
+    const { syncReqToEnqueue, duplicateSyncReq } = getNewOrExistingSyncReq({
+      userWallet,
+      secondaryEndpoint,
+      primaryEndpoint: thisContentNodeEndpoint,
+      syncType,
+      syncMode: syncModeResp
+    })
+    if (duplicateSyncReq && !_.isEmpty(duplicateSyncReq)) {
```

Comment:
nit: shouldn't need the first `duplicateSyncReq` check since I believe isEmpty returns false for null and undefined

Reply:
i took it from your code though 🙃 https://github.com/AudiusProject/audius-protocol/pull/3478/files#diff-6dc4978ddcba7af995bf5124a22398c62414e361915a7773087003f8301ec53cL222

but yeah you right i'll remove

Reply:
well, it was existing code, idk who wrote it

Reply:
oh lmao no idea why I did that. I just confirmed that `_.isEmpty()` returns true for null and undefined

Reply:
all good hahah, i updated